File: leetcode_2270/NumberOfWaysToSplitArray.java
class  NumberOfWaysToSplitArray{
    public int waysToSplitArray(int[] nums) {
        long leftSum = 0, rightSum = 0;
        for(int n : nums)
            rightSum += n;
        int validSplit = 0;

        for(int i=0; i<nums.length - 1; i++)
        {
            leftSum += nums[i];
            rightSum -= nums[i];
            if(leftSum >= rightSum)
                validSplit++;
        }
        return validSplit;
    }

    public static void main(String[] args) {
        NumberOfWaysToSplitArray obj = new NumberOfWaysToSplitArray();
        int[] nums = {10,4,-8,7};
        System.out.println(obj.waysToSplitArray(nums));
    }
}

File: leetcode_2270/NumberOfWaysToSplitArray.py
class  NumberOfWaysToSplitArray:
    def waysToSplitArray(self, nums: list[int]) -> int:
        leftSum, rightSum = 0, sum(nums)
        validSplit = 0
        
        for i in range(len(nums) - 1):
            leftSum += nums[i]
            rightSum -= nums[i]
            if leftSum >= rightSum:
                validSplit += 1
        return validSplit
    
def main():
    v = NumberOfWaysToSplitArray()
    nums = [10, 4, -8, 7]
    print(v.waysToSplitArray(nums))

if __name__ == "__main__":
    main()